cli/bpmetadata/path.go

```diff
@@ -0,0 +1,85 @@
+package bpmetadata
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+)
+
+const (
+	regexEx  = ".*/(examples/.*)"
+	regexMod = ".*/(modules/.*)"
+)
+
+func isPathValid(path string) (bool, error) {
+	_, err := os.ReadFile(path)
+	if err != nil {
+		return false, fmt.Errorf("unable to read file at the provided path: %v", err)
+	}
+
+	return true, nil
+}
+
+func getExamples(configPath string) ([]BlueprintMiscContent, error) {
+	re := regexp.MustCompile(regexEx)
+	return getDirPaths(configPath, re)
+}
+
+func getModules(configPath string) ([]BlueprintMiscContent, error) {
+	re := regexp.MustCompile(regexMod)
+	return getDirPaths(configPath, re)
+}
+
+func getDirPaths(configPath string, re *regexp.Regexp) ([]BlueprintMiscContent, error) {
```

Comment:
Lets add a comment as to what this function is doing. I assumed it was collecting dirs in `configPath` but seems like it maybe doing more.

Reply:
added